test/e2e_node/hugepages_test.go

```diff
@@ -117,43 +118,89 @@ func makePodToVerifyHugePages(baseName string, hugePagesLimit resource.Quantity,
 	return pod
 }
 
-// configureHugePages attempts to allocate hugepages of the specified size
-func configureHugePages(hugepagesSize int, hugepagesCount int) error {
-	// Compact memory to make bigger contiguous blocks of memory available
-	// before allocating huge pages.
-	// https://www.kernel.org/doc/Documentation/sysctl/vm.txt
-	if _, err := os.Stat("/proc/sys/vm/compact_memory"); err == nil {
-		if err := exec.Command("/bin/sh", "-c", "echo 1 > /proc/sys/vm/compact_memory").Run(); err != nil {
-			return err
-		}
-	}
-
+// reserveHugePages just writes appropriate value into sysfs
+func reserveHugePages(hugepagesSize int, hugepagesCount int) error {
 	// Reserve number of hugepages
 	// e.g. /bin/sh -c "echo 5 > /sys/kernel/mm/hugepages/hugepages-2048kB/vm.nr_hugepages"
 	command := fmt.Sprintf("echo %d > %s-%dkB/%s", hugepagesCount, hugepagesDirPrefix, hugepagesSize, hugepagesCapacityFile)
 	if err := exec.Command("/bin/sh", "-c", command).Run(); err != nil {
 		return err
 	}
 
-	// verify that the number of hugepages was updated
-	// e.g. /bin/sh -c "cat /sys/kernel/mm/hugepages/hugepages-2048kB/vm.nr_hugepages"
-	command = fmt.Sprintf("cat %s-%dkB/%s", hugepagesDirPrefix, hugepagesSize, hugepagesCapacityFile)
-	outData, err := exec.Command("/bin/sh", "-c", command).Output()
-	if err != nil {
-		return err
+	if hugepagesCount > 0 {
+		// verify that the number of hugepages was updated
+		// e.g. /bin/sh -c "cat /sys/kernel/mm/hugepages/hugepages-2048kB/vm.nr_hugepages"
+		command = fmt.Sprintf("cat %s-%dkB/%s", hugepagesDirPrefix, hugepagesSize, hugepagesCapacityFile)
+		outData, err := exec.Command("/bin/sh", "-c", command).Output()
+		if err != nil {
+			return err
+		}
+
+		numHugePages, err := strconv.Atoi(strings.TrimSpace(string(outData)))
+		if err != nil {
+			return err
+		}
+
+		framework.Logf("Hugepages total is set to %v", numHugePages)
+		if numHugePages == hugepagesCount {
+			return nil
+		}
+
+		return fmt.Errorf("expected hugepages %v, but found %v", hugepagesCount, numHugePages)
 	}
+	return nil
+}
 
-	numHugePages, err := strconv.Atoi(strings.TrimSpace(string(outData)))
+func traceMemInfo() {
+	file, err := os.Open("/proc/meminfo")
 	if err != nil {
-		return err
+		framework.Logf("Can't read meminfo %v", err)
+		return
 	}
+	defer file.Close()
 
-	framework.Logf("Hugepages total is set to %v", numHugePages)
-	if numHugePages == hugepagesCount {
-		return nil
+	scanner := bufio.NewScanner(file)
+	// optionally, resize scanner's capacity for lines over 64K, see next example
```

Comment:
do you want to just print the whole `/proc/meminfo`, right? If so why not just `os.ReadFile` ?

Reply:
yes, it prints whole /proc/meminfo,
thank you for suggestion, I'll consider os.ReadFile too.